Applications/Spire/Source/Ui/MenuButton.cpp

```diff
@@ -0,0 +1,120 @@
+#include "Spire/Ui/MenuButton.hpp"
+#include <QCoreApplication>
+#include <QKeyEvent>
+#include "Spire/Ui/Button.hpp"
+#include "Spire/Ui/Layouts.hpp"
+#include "Spire/Ui/OverlayPanel.hpp"
+
+using namespace Spire;
+using namespace Spire::Styles;
+
+MenuButton::MenuButton(QWidget* body, MenuBuilder menu_builder, QWidget* parent)
+    : QWidget(parent),
+      m_timer(this),
+      m_is_mouse_down_on_button_with_menu(false),
+      m_is_mouse_down_on_button_without_menu(false) {
+  setFocusPolicy(Qt::StrongFocus);
+  enclose(*this, *body);
+  m_menu = menu_builder(this);
```

Comment:
Can you remove the use of the ```menu_builder```?

Reply:
Do you mean to update the API?

Reply:
@kranar 
I updated the API:

1. Removed menu_builder. An OverlayPanel reference will be used to construct the MenuButton.
2. Updated the body pointer to the body reference in the constructor.

[API](https://gist.github.com/shaobinspire/752f960eb10dca704b4f8d8618ca5a8d)

Reply:
Much better.